commit c36d91bd4ebf767f310f7cea88d61d1c15f53ddf
Author: djm@openbsd.org <djm@openbsd.org>
Date:   Fri Jul 22 03:39:13 2016 +0000

    upstream commit

    move debug("%p", key) to before key is free'd; probable
    undefined behaviour on strict compilers; reported by Jakub Jelen bz#2581

    Upstream-ID: 767f323e1f5819508a0e35e388ec241bac2f953a

--- openssh-3.6.1p2/monitor.c.orig	2003-04-29 09:12:08 +0000
+++ openssh-3.6.1p2/monitor.c	2016-08-23 19:53:06 +0000
@@ -807,6 +807,10 @@ mm_answer_keyallowed(int socket, Buffer
 			break;
 		}
 	}
+
+	debug3("%s: key %p is %s",
+	    __func__, key, allowed ? "allowed" : "disallowed");
+
 	if (key != NULL)
 		key_free(key);
 
@@ -822,9 +826,6 @@ mm_answer_keyallowed(int socket, Buffer
 		hostbased_chost = chost;
 	}
 
-	debug3("%s: key %p is %s",
-	    __func__, key, allowed ? "allowed" : "disallowed");
-
 	buffer_clear(m);
 	buffer_put_int(m, allowed);
 	buffer_put_int(m, forced_command != NULL);
